sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -1762,37 +1765,34 @@ public Mono<Response<ShareFileInfo>> setPropertiesWithResponse(long newFileSize,
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<ShareFileInfo>> setPropertiesWithResponse(ShareFileSetPropertiesOptions options) {
         try {
+            StorageImplUtils.assertNotNull("options", options);
```

Comment:
yep, it would fail. i found it through testing. this is a small part of a much larger problem through files - inconsistencies in how we handle options bags. at some point I need to go through and add these asserts to several other places - especially to those that have a mandatory parameter in the options bag.